src/scheduler/job_info.cpp

```diff
@@ -1234,6 +1236,8 @@ query_job(struct batch_status *job, server_info *sinfo, schd_error *err)
 			else
 				resresv->qrank = -1;
 		}
+		else if (!strcmp(attrp->name, ATTR_server_inst_id))
+			resresv->job->svr_inst_id = string_dup(attrp->value);
```

Comment:
do a NULL check here as well?

Reply:
No need to do a NULL check to just print a message.  That happens inside string_dup().  Although error recovery here is kind of important.